include/seqan/graph_types/graph_impl_directed.h

```diff
@@ -74,55 +74,112 @@ class Graph<Directed<TCargo, TSpec> >
         }
 
-        ~Graph() {
-            SEQAN_CHECKPOINT
+        Graph const & operator = (Graph const & _other)
+        {
+            if (this == &_other)
+                return *this;
+
+            clear(*this);
+            data_allocator = _other.data_allocator;
+            _copyGraph(*this, _other);
+            return *this;
+        }
+};
+
+template <typename TCargo>
+class Graph<Directed<TCargo, WithSourceId> >
+{
+    public:
+        typedef typename VertexIdHandler<Graph>::Type TVertexIdManager_;
+        typedef typename EdgeIdHandler<Graph>::Type TEdgeIdManager_;
+        typedef typename EdgeType<Graph>::Type TEdgeStump_;
+        typedef Allocator<SinglePool<sizeof(TEdgeStump_)> > TAllocator_;
+
+        String<TEdgeStump_*> data_vertex;            // Pointers to EdgeStump lists
+        String<TEdgeStump_*> data_vertex_in;         
+        TVertexIdManager_ data_id_managerV;
+        TEdgeIdManager_ data_id_managerE;
+        TAllocator_ data_allocator;
+
+//____________________________________________________________________________
+
+        Graph() {}
+
+        ~Graph()
+        {
             clear(*this);
         }
 
         Graph(Graph const & _other) :
             data_allocator(_other.data_allocator)
         {
-            SEQAN_CHECKPOINT
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
         }
 
-        Graph const& operator = (Graph const & _other) {
-            SEQAN_CHECKPOINT
-            if (this == &_other) return *this;
+        Graph const & operator = (Graph const & _other)
+        {
+            if (this == &_other)
+                return *this;
+
             clear(*this);
             data_allocator = _other.data_allocator;
-            _copyGraph(_other, *this);
+            _copyGraph(*this, _other);
             return *this;
         }
 };
 
-
 //////////////////////////////////////////////////////////////////////////////
 // INTERNAL FUNCTIONS
 //////////////////////////////////////////////////////////////////////////////
 
 //////////////////////////////////////////////////////////////////////////////
 
-template<typename TCargo, typename TSpec>
-inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*>&
-_getVertexString(Graph<Directed<TCargo, TSpec> > const& g) {
-    SEQAN_CHECKPOINT
+template <typename TCargo, typename TSpec>
+inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*> &
+_getVertexString(Graph<Directed<TCargo, TSpec> > const & g)
+{
     typedef Graph<Directed<TCargo, TSpec> > TGraph;
     typedef typename EdgeType<TGraph>::Type TEdgeStump;
     return const_cast<String<TEdgeStump*>&>(g.data_vertex);
 }
 
+//////////////////////////////////////////////////////////////////////////////
+
+template <typename TCargo, typename TSpec>
+inline typename Size<Graph<Directed<TCargo, WithSourceId> > >::Type
+_getVertexStringSize(Graph<Directed<TCargo, TSpec> > const & g)
```

Comment:
As far as I can see this function is only used in _copyGraph, which is overloaded for DirectedGraphs and your specialisation?
Why bother to add a private function that returns just the size. Just call length(g.data_vertex_in) and length(g.data_vertex), where you need it.